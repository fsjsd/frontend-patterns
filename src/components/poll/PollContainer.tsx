import React, { Suspense } from 'react'
import { ContentContainer } from '../../ux/ContentContainer'
import Loading from '../../ux/Loading'
const Poll = React.lazy(() => import('./Poll'))

const PollContainer = () => {
  return (<>
    <ContentContainer
      title="Poll"
      codeLink="/features/poll"
      markDownPromise={import('./requirements.md')}
    >
      <Suspense fallback={<Loading />}>
        <div role="main">
          <Poll />
        </div>
      </Suspense>
    </ContentContainer>
  </>
  )
}

export default PollContainer