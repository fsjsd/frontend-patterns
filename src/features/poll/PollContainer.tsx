import React, { Suspense } from 'react'
import ContentHeader, { ContentHeaderLabel } from '../../ux/ContentHeader'
import { ContentWrapper } from '../../ux/ContentWrapper'
import Loading from '../../ux/Loading'
const Poll = React.lazy(() => import('./Poll'))

const PollContainer = () => {
  return (<>
    <ContentHeader>
      <ContentHeaderLabel>Poll</ContentHeaderLabel>
    </ContentHeader>
    <ContentWrapper>
      <Suspense fallback={<Loading />}>
        <div role="main">
          <Poll />
        </div>
      </Suspense>
    </ContentWrapper>
  </>
  )
}

export default PollContainer