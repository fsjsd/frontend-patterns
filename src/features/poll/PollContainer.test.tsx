import React from 'react'
import PollContainer from "./PollContainer";
import { render, waitFor } from '@testing-library/react'

describe("Poll", () => {
  it("renders and matches snapshot", async () => {
    const { container, queryByRole } = render(<PollContainer />)
    await waitFor(() => expect(queryByRole("main")).toBeInTheDocument());
    expect(container).toMatchSnapshot();

  })
})