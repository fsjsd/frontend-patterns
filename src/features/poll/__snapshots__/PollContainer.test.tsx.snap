// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ dtMXGe"
  >
    <div
      class="sc-gsnTZi dkPlsI"
    >
      Poll
    </div>
  </div>
  <div
    class="sc-dkzDqf eGEKCX"
  >
    <div
      role="main"
      style=""
    >
      <div
        class="sc-eCYdqJ jGnNot"
      >
        Coming soon
      </div>
    </div>
  </div>
</div>
`;
